pkg/pubsub/adapter/adapter.go

```diff
@@ -167,10 +176,15 @@ func (a *Adapter) receive(ctx context.Context, event cloudevents.Event, resp *cl
 		ResourceGroup: a.ResourceGroup,
 	}
 
-	// a.Sink is likely not exactly what we want...
-	ctx, err := tracing.AddSpanFromTraceparentAttribute(ctx, a.Sink, event)
-	if err != nil {
-		logger.Infow("Unable to attach tracing to context", zap.Error(err))
+	var err error
+	// If the resource group is the Channel one, then we attach the span from the traceparent attribute.
+	// Otherwise, it means it's a Source, thus it does not have traceparent.
+	if a.ResourceGroup == channelGVR.GroupResource().String() {
+		// a.Sink is likely not exactly what we want...
+		ctx, err = tracing.AddSpanFromTraceparentAttribute(ctx, a.Sink, event)
+		if err != nil {
+			logger.Debugw("Unable to attach tracing to context", zap.Error(err))
```

Comment:
Please leave it as a warning, I want to be aware of it happening.

Reply:
warning it is then! thanks!